lib/memstore.ts

```diff
@@ -39,24 +39,30 @@ import {
 import { Store } from './store'
 import { getCustomInspectSymbol, getUtilInspect } from './utilHelper'
 
-export class MemoryCookieStore extends Store {
-  override synchronous: boolean
-  idx: {
-    [domain: string]: {
-      [path: string]: {
-        [key: string]: Cookie
-      }
+export type MemoryCookieStoreIndex = {
+  [domain: string]: {
+    [path: string]: {
+      [key: string]: Cookie
     }
   }
+}
+
+export class MemoryCookieStore extends Store {
+  override synchronous: boolean
+  idx: MemoryCookieStoreIndex
 
   constructor() {
     super()
     this.synchronous = true
-    this.idx = Object.create(null)
+    this.idx = Object.create(null) as MemoryCookieStoreIndex
     const customInspectSymbol = getCustomInspectSymbol()
     if (customInspectSymbol) {
-      // @ts-ignore
-      this[customInspectSymbol] = this.inspect
+      Object.defineProperty(this, customInspectSymbol, {
+        value: this.inspect.bind(this),
```

Comment:
Why use a bound function rather than the original one?

Reply:
it was the only way to eliminate the `@ts-ignore`